tests/includes/juju.sh

```diff
@@ -266,6 +266,11 @@ pre_bootstrap() {
 		version=$(juju_version)
 		export BOOTSTRAP_ADDITIONAL_ARGS="${BOOTSTRAP_ADDITIONAL_ARGS:-} --agent-version=${version}"
 	fi
+
+	if [[ -n ${SHORT_GIT_COMMIT:-} ]] && [[ ${BOOTSTRAP_PROVIDER:-} == "aws" ]] && [[ ${MODEL_ARCH:-} != "amd64" ]];  then
```

Comment:
Why limit to aws and amd64? The binaries are published for all arches. There's nothing provider specific about the binaries either. 
This will fail if run locally since SHORT_GIT_COMMIT won't be set.
We also need to consider k8s and set the caas-image repo param